pyrobosim/pyrobosim/planning/pddlstream/utils.py

```diff
@@ -156,6 +157,11 @@ def pddlstream_solution_to_plan(solution, robot):
                 arg = act_pddl.args[3]
                 if isinstance(arg, Pose):
                     act.pose = arg
+            # If a grasp pose is specified for the pick action, use that instead.
+            if act.type == "pick":
+                for arg in act_pddl.args:
+                    if isinstance(arg, Grasp):
+                        act.pose = arg.origin
```

Comment:
Notice with this logic that it's not the entire grasp object that makes it to the action plan, but rather the grasp pose.

You lose all the metadata like front vs. side grasp, etc, but I don't think it should matter?

It would take some expansion of the Action Python type + ROS messages to change this, but if deemed worthwhile it can be done!

Reply:
I don't foresee TOP/FRONT metadata being important any time soon...

Reply:
I don't foresee TOP/FRONT metadata being important any time soon...